spec/src/modules/assistant.js

```diff
@@ -0,0 +1,271 @@
+      expect(() => createAssistantUrl('testIntent', {}, defaultOptions)).throw('parameters.domain is a required parameter of type string');
+    });
+
+    it('should correctly construct a URL with minimal valid inputs', () => {
+      const intent = 'testIntent';
+      const url = createAssistantUrl(intent, defaultParameters, defaultOptions);
+
+      expect(url).contain('https://assistant.cnstrc.com/v1/intent/');
+      expect(url).contain(`intent/${encodeURIComponentRFC3986(intent)}`);
+
+      const requestedUrlParams = qs.parse(url.split('?')?.[1]);
+
+      expect(requestedUrlParams).to.have.property('key');
+      expect(requestedUrlParams).to.have.property('i');
+      expect(requestedUrlParams).to.have.property('s');
+      expect(requestedUrlParams).to.have.property('c').to.equal(clientVersion);
+      expect(requestedUrlParams).to.have.property('_dt');
+
+    });
+
+    it('should clean and encode parameters correctly', () => {
+      const intentWithSpaces = 'test Intent';
+      const url = createAssistantUrl(
+        intentWithSpaces,
+        { ...defaultParameters },
+        defaultOptions,
+      );
+
+      // Ensure spaces are encoded and parameters are cleaned
+      expect(url).not.contain(' ');
+      expect(url).contain(encodeURIComponentRFC3986(intentWithSpaces));
+    });
+  });
+
+  // setupEventListeners util Tests
+  describe('Test setupEventListeners', () => {
+    let mockEventSource;
+    let mockStreamController;
+
+    beforeEach(() => {
+      mockEventSource = {
+        addEventListener: sinon.stub(),
+        close: sinon.stub(),
+      };
+
+      mockStreamController = {
+        enqueue: sinon.stub(),
+        close: sinon.stub(),
+        error: sinon.stub(),
+      };
+    });
+
+    afterEach(() => {
+      sinon.restore(); // Restore all mocks
+    });
+
+    it('should set up event listeners for all event types', () => {
+      const eventTypes = Assistant.EventTypes;
+
+      setupEventListeners(mockEventSource, mockStreamController, eventTypes);
+
+      expect(mockEventSource.addEventListener.calledWith('start')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('group')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('search_result')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('article_reference')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('recipe_info')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('recipe_instructions')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('server_error')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('image_meta')).to.be.true;
+      expect(mockEventSource.addEventListener.calledWith('end')).to.be.true;
+    });
+
+    it('should enqueue event data into the stream', (done) => {
+      const eventTypes = Assistant.EventTypes;
+      const eventType = 'search_result';
```

Comment:
Would it make sense to set `eventType = Assistant.EventTypes.SEARCH_RESULT` here?

Reply:
Done!